From 2fb4fdf641e67e49b87a3524038a694c8dd0ba4e Mon Sep 17 00:00:00 2001
From: Vicente Bergas <vicencb@gmail.com>
Date: Fri, 9 Oct 2015 23:35:20 +0200
Subject: [PATCH] ecore_fb_private.h: define EAPI before use

From [1]:
When running terminology, a message appears in eina_module_load with:
could not dlopen("/usr/lib/ecore_evas/engines/fb/v-1.15/module.so",
Error relocating /usr/lib/ecore_evas/engines/fb/v-1.15/module.so:
ecore_fb_ts_shutdown: symbol not found): RTLD_NOW
It seems like the EAPI macro has no effect...

A patch from Ross Vandegrift has been posted on enlightenment mailing
list [2], but it's not yet an upstream patch.

[1] http://sourceforge.net/p/enlightenment/mailman/message/34493376
[2] http://sourceforge.net/p/enlightenment/mailman/message/34492801

Signed-off-by: Vicente Bergas <vicencb@gmail.com>
[Romain:
  - Add a commit log
  - Add a link to the enlightenment mailing list
  - Add Vicente's SoB line]
Signed-off-by: Romain Naour <romain.naour@openwide.fr>
---
 src/lib/ecore_fb/ecore_fb_private.h | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/src/lib/ecore_fb/ecore_fb_private.h b/src/lib/ecore_fb/ecore_fb_private.h
index f7dc0c6..f54c8d2 100644
--- a/src/lib/ecore_fb/ecore_fb_private.h
+++ b/src/lib/ecore_fb/ecore_fb_private.h
@@ -33,6 +33,20 @@
 
 #include <Ecore_Fb.h>
 
+#ifdef EAPI
+# undef EAPI
+#endif
+
+#ifdef __GNUC__
+# if __GNUC__ >= 4
+#  define EAPI __attribute__ ((visibility("default")))
+# else
+#  define EAPI
+# endif
+#else
+# define EAPI
+#endif
+
 /* ecore_fb_li.c */
 struct _Ecore_Fb_Input_Device
 {
@@ -92,4 +106,7 @@ void ecore_fb_vt_shutdown(void);
 #define TS_GET_CAL 0x8014660a
 #endif
   
+#undef EAPI
+#define EAPI
+
 #endif
-- 
2.4.3

